crates/shim-sgx/src/handler/mod.rs

```diff
@@ -77,17 +77,22 @@ pub struct Handler<'a> {
 }
 
 impl<'a> Handler<'a> {
-    /// Acknowledge pages committed by the host with ENCLS[EAUG].
-    fn accept_mmap(&mut self, region: Region) {
-        assert!(region.start < region.end);
-        let length = region.end - region.start;
-        let secinfo = Class::Regular.info(Flags::READ | Flags::WRITE | Flags::EXECUTE);
+    /// Acknowledge ENCLS[EAUG].
+    fn accept_pending(&mut self, addr: usize, length: c_size_t, prot: c_int) {
+        let flags = Flags::from_bits_truncate((prot | PROT_READ) as u8);
         let zero = PageAddr::from_start_address(VirtAddr::new(ZERO.as_ptr() as u64)).unwrap();
-        for i in 0..length.items() {
-            let virt_addr = VirtAddr::new((region.start.raw() + i * Page::SIZE) as u64);
-            let page_addr = PageAddr::from_start_address(virt_addr).unwrap();
-            // FIXME: handle with `.unwrap()` once `SecInfo` implements `Debug` trait.
-            if secinfo.accept_copy(page_addr, zero).is_err() {
+        for i in 0..length / Page::SIZE {
+            let addr = PageAddr::from_start_address(VirtAddr::new((addr + i * Page::SIZE) as u64))
+                .unwrap();
```

Comment:
This is an example of a wrapping caused by a complex line. This should always be avoided by splitting the complex line into two simpler statements. Please do this everywhere.

Reply:
i did :-)